Apps/Sandcastle/gallery/development/Custom Primitive.html

```diff
@@ -0,0 +1,327 @@
+        padding-top: 2px;
+        padding-bottom: 2px;
+      }
+    </style>
+    <div id="cesiumContainer" class="fullSize"></div>
+    <div id="loadingOverlay"><h1>Loading...</h1></div>
+
+    <script id="cesium_sandcastle_script">
+      function startup(Cesium) {
+        "use strict";
+        //Sandcastle_Begin
+
+        var viewer = new Cesium.Viewer("cesiumContainer");
+
+        /**
+         * Simple example of writing a Primitive from scratch. This
+         * primitive displays a procedurally-generated surface at a given
+         * position on the globe.
+         */
+        function CustomPrimitive(cartographicPosition, context) {
+          this.show = true;
+
+          this.drawCommand = undefined;
+
+          // number of faces wide. There are resolution + 1 vertices.
+          this.faceResolution = 200;
+
+          // Compute a model matrix that puts the surface at a specific point on
+          // the globe.
+          this.cartographicPosition = cartographicPosition;
+          var cartesianPosition = Cesium.Cartographic.toCartesian(
+            cartographicPosition,
+            Cesium.Ellipsoid.WGS84,
+            new Cesium.Cartesian3()
+          );
+          this.modelMatrix = Cesium.Transforms.eastNorthUpToFixedFrame(
+            cartesianPosition,
+            Cesium.Ellipsoid.WGS84,
+            new Cesium.Matrix4()
+          );
+
+          this.halfWidthMeters = 10000;
+          this.boundingSphere = new Cesium.BoundingSphere(
+            cartesianPosition,
+            this.halfWidthMeters * Math.SQRT2
+          );
+
+          this.time = undefined;
+
+          initialize(this, context);
+        }
+
+        /**
+         * generate a quad that's subdivided into faceResolution x faceResolution quads.
+         * The vertices will be adjusted in the vertex shader.
+         */
+        function generateQuad(faceResolution, halfWidthMeters) {
+          var vertexResolution = faceResolution + 1;
+          var vertexCount = vertexResolution * vertexResolution;
+          var componentsPerVertex = 3;
+          var vertices = new Float32Array(vertexCount * componentsPerVertex);
+          for (var i = 0; i < vertexResolution; i++) {
+            for (var j = 0; j < vertexResolution; j++) {
+              var u = i / (vertexResolution - 1);
+              var v = j / (vertexResolution - 1);
+              var index = i * vertexResolution + j;
+              var x = halfWidthMeters * (2 * u - 1);
+              var y = halfWidthMeters * (2 * v - 1);
+              var z = 0;
+
+              vertices[index * componentsPerVertex] = x;
+              vertices[index * componentsPerVertex + 1] = y;
+              vertices[index * componentsPerVertex + 2] = z;
+            }
+          }
+          return vertices;
+        }
+
+        /**
+         * tesellate a big square region into faceResolution x faceResolution quads
+         */
+        function tesellate(faceResolution) {
```

Comment:
same idea here, though this name makes more sense to me. maybe `generateIndices`?